internal/pkg/manifest/workload_test.go

```diff
@@ -649,7 +651,117 @@ func TestNetworkConfig_IsEmpty(t *testing.T) {
 	}
 }
 
+func TestSecurityGroupsConfig_UtilityFuncForSecurityGroups(t *testing.T) {
+	testCases := map[string]struct {
+		data                 SecurityGroupsIDsOrConfig
+		wantedSecurityGroups []string
+	}{
+		"nil returned when no security groups are specified": {
+			data:                 SecurityGroupsIDsOrConfig{},
+			wantedSecurityGroups: nil,
+		},
+		"security groups in map are returned": {
+			data: SecurityGroupsIDsOrConfig{
+				AdvancedConfig: SecurityGroupsConfig{
+					SecurityGroups: []string{"group", "group1"},
+				},
+			},
+			wantedSecurityGroups: []string{"group", "group1"},
+		},
+		"nil returned when security groups in map are empty": {
+			data: SecurityGroupsIDsOrConfig{
+				AdvancedConfig: SecurityGroupsConfig{
+					SecurityGroups: []string{},
+				},
+			},
+			wantedSecurityGroups: nil,
+		},
+		"security groups in array are returned": {
+			data: SecurityGroupsIDsOrConfig{
+				IDs: []string{"123", "45"},
+			},
+			wantedSecurityGroups: []string{"123", "45"},
+		},
+		"nil returned when security groups in array are empty": {
+			data: SecurityGroupsIDsOrConfig{
+				IDs: []string{},
+			},
+			wantedSecurityGroups: nil,
+		},
+	}
+
+	for name, tc := range testCases {
+		t.Run(name, func(t *testing.T) {
+			// WHEN
+			sgs := tc.data.GetIDs()
+
+			// THEN
+			require.Equal(t, tc.wantedSecurityGroups, sgs)
+		})
+	}
+}
+
+func TestSecurityGroupsConfig_UtilityFuncForDefaultSecurityGroup(t *testing.T) {
+	var (
+		falseValue = false
+		trueValue  = true
+	)
+	testCases := map[string]struct {
+		data              SecurityGroupsIDsOrConfig
+		IsDefaultSGDenied bool
+	}{
+		"default security group is applied when no vpc security config is present": {
+			data:              SecurityGroupsIDsOrConfig{},
+			IsDefaultSGDenied: false,
+		},
+		"default security group is applied when deny_default is not specified in SG config": {
+			data: SecurityGroupsIDsOrConfig{
+				AdvancedConfig: SecurityGroupsConfig{
+					SecurityGroups: []string{"1"},
+				},
+			},
+			IsDefaultSGDenied: false,
+		},
+		"default security group is applied when deny_default is false in SG config": {
+			data: SecurityGroupsIDsOrConfig{
+				AdvancedConfig: SecurityGroupsConfig{
+					SecurityGroups: []string{"1"},
+					DenyDefault:    &falseValue,
```

Comment:
nit: similar to the other `aws.Bool` nits!
```suggestion
					DenyDefault:    aws.Bool(false),
```